sklearn/utils/_print_versions.py

```diff
@@ -0,0 +1,156 @@
+"""
+Utility methods to print system info for debugging
+
+adapted from :func:`pandas.show_versions`
+"""
+# License: BSD 3 clause
+
+import os
+import platform
+import sys
+import struct
+import subprocess
+import codecs
+import locale
+import importlib
+
+from .._build_utils import get_blas_info
+
+
+def get_sys_info():
+    """System information
+
+    Return
+    ------
+    sys_info : dict
+        system and Python version information
+
+    """
+
+    blob = []
+
+    # get full commit hash
+    commit = None
+    if os.path.isdir(".git") and os.path.isdir("sklearn"):
+        try:
+            pipe = subprocess.Popen('git log --format="%H" -n 1'.split(" "),
+                                    stdout=subprocess.PIPE,
+                                    stderr=subprocess.PIPE)
+            so, serr = pipe.communicate()
+        except:
+            pass
+        else:
+            if pipe.returncode == 0:
+                commit = so
+                try:
+                    commit = so.decode('utf-8')
+                except ValueError:
+                    pass
+                commit = commit.strip().strip('"')
+
+    blob.append(('commit', commit))
+
+    try:
+        (sysname, nodename, release,
+         version, machine, processor) = platform.uname()
+        blob.extend([
+            ("python", '.'.join(map(str, sys.version_info))),
+            ("python-bits", struct.calcsize("P") * 8),
+            ("OS", "{sysname}".format(sysname=sysname)),
+            ("OS-release", "{release}".format(release=release)),
+            ("machine", "{machine}".format(machine=machine)),
+            ("processor", "{processor}".format(processor=processor)),
+            ("byteorder", "{byteorder}".format(byteorder=sys.byteorder)),
+            ("LC_ALL", "{lc}".format(lc=os.environ.get('LC_ALL', "None"))),
+            ("LANG", "{lang}".format(lang=os.environ.get('LANG', "None"))),
+            ("LOCALE", '.'.join(map(str, locale.getlocale()))),
+        ])
+    except:
+        pass
+
+    return dict(blob)
+
+
+def get_deps_info():
+    """Overview of the installed version of main dependencies
+
+    Returns
+    -------
+    deps_info: dict
+        version information on relevant Python libraries
+
+    """
+    deps = [
+        # (MODULE_NAME, f(mod) -> mod version)
+        ("pip", lambda mod: mod.__version__),
+        ("setuptools", lambda mod: mod.__version__),
+        ("numpy", lambda mod: mod.version.version),
+        ("scipy", lambda mod: mod.version.version),
+        ("Cython", lambda mod: mod.__version__),
+        ("pandas", lambda mod: mod.__version__),
+        ("matplotlib", lambda mod: mod.__version__),
```

Comment:
is pandas necessary as it's not a requirement for scikit-learn ?
And shouldn't we add scikit-learn's version ?

Reply:
yeah, probably nice to have 😄

Reply:
pandas is often used together with matplotlib and would be nice to have to debug `ColumnTransformer` etc issues.